### PR TITLE
credentials/alts: Update ALTS "New" APIs

### PR DESCRIPTION
**Pull request by cesarghali**
Change the `alts.NewClient` API to take a struct containing all client-side options instead of passing them using parameters. This allows adding more options in the future without breaking the existing API or having to create a new one.

This API also changes `alts.NewClient` and `alts.NewServer` to `alts.NewClientCreds` and `alts.NewServerCreds`, respectively.

This API is only used by the `interop/alts/client/client.go`.

### CODE REVIEW

File: credentials/alts/alts.go [hunk 1]
```diff
@@ -91,6 +91,19 @@ type AuthInfo interface {
...
+// ClientOptions contains the client-side options of an ALTS channel. These
+// options will be passed to the underlying ALTS handshaker.
+type ClientOptions struct {
+	// TargetServiceAccounts contains a list of expected target service
+	// accounts.
+	TargetServiceAccounts []string
+}
+
+// DefaultClientOptions creates a default (empty) ClientOptions.
+func DefaultClientOptions() *ClientOptions {
```
Comment (dfawley) [inline credentials/alts/alts.go]: This should not be necessary; your zero-value defaults should all be sane (and I see you do end up with the zero value, so this is true).

File: credentials/alts/alts.go [hunk 2]
```diff
@@ -91,6 +91,19 @@ type AuthInfo interface {
 	PeerRPCVersions() *altspb.RpcProtocolVersions
 }
 
+// ClientOptions contains the client-side options of an ALTS channel. These
+// options will be passed to the underlying ALTS handshaker.
+type ClientOptions struct {
+	// TargetServiceAccounts contains a list of expected target service
+	// accounts.
+	TargetServiceAccounts []string
```
Comment (dfawley) [inline credentials/alts/alts.go]: Is this always (or nearly always) supposed to be non-empty?  Or is it common to omit it?

If it's commonly used, it probably should be left as a parameter.

Comment (cesarghali) [reply credentials/alts/alts.go]: This is not commonly used. I'm expecting callers will set this in very few situations.

File: credentials/alts/alts.go (hunk 1 above)
Comment (cesarghali) [reply credentials/alts/alts.go]: Done!

**Comment by cesarghali** [comment]
Thanks for the comments Doug.

Regarding the flag, there was a discussion about it in #1865. I'm happy to discuss it more offline and we do further changes in a separate PR.

**Pull request merged**